src/napari_spatialdata/_scatterwidgets.py

```diff
@@ -23,6 +29,81 @@
 ]
 
 
+class SelectFromCollection:
+    """
+    Select indices from a matplotlib collection using `LassoSelector`.
+
+    Selected indices are saved in the `ind` attribute. This tool fades out the
+    points that are not part of the selection (i.e., reduces their alpha
+    values). If your collection has alpha < 1, this tool will permanently
+    alter the alpha values.
+
+    Note that this tool selects collection objects based on their *origins*
+    (i.e., `offsets`).
+
+    Parameters
+    ----------
+    ax : `~matplotlib.axes.Axes`
+        Axes to interact with.
+    collection : `matplotlib.collections.Collection` subclass
+        Collection you want to select from.
+    alpha_other : 0 <= float <= 1
+        To highlight a selection, this tool sets all selected points to an
+        alpha value of 1 and non-selected points to *alpha_other*.
+    """
+
+    def __init__(
+        self,
+        viewer: Viewer,
+        model: ImageModel,
+        ax: Axes,
+        collection: Collection,
+        data: List[NDArrayA],
+        alpha_other: float = 0.3,
+    ):
+        self.viewer = viewer
+        self.model = model
+        self.canvas = ax.figure.canvas
+        self.collection = collection
+        self.alpha_other = alpha_other
+        self.exported_data = None
+        self.data = data
+        self.axes = ax
+
+        self.xys = collection.get_offsets()
+        self.Npts = len(self.xys)
+
+        # Ensure that we have separate colors for each object
+        self.fc = collection.get_facecolors()
+
+        if len(self.fc) == 0:
+            raise ValueError("Collection must have a facecolor")
+        elif len(self.fc) == 1:
+            self.fc = np.tile(self.fc, (self.Npts, 1))
+
+        self.selector = LassoSelector(ax, onselect=self.onselect)
+
+        self.ind: Optional[NDArrayA] = None
+
+    def export(self, adata: AnnData) -> None:
+        adata.obs["0_LASSO_SELECTED"] = self.exported_data
```

Comment:
I think this should containe the info of the active anndata that was used to crop (e.g. it's a lasso of which anndata object?). The name could then be something like: `{layer_name}_LASSO_SELECTED`

Reply:
should I still add "0_ "in front so that it turns up in the top?
<img width="499" alt="image" src="https://user-images.githubusercontent.com/22676591/215287275-79f88ee8-6cef-4314-b6c1-2774908d93ad.png">

Reply:
looks like this with the latest commit 
<img width="799" alt="image" src="https://user-images.githubusercontent.com/22676591/215290928-9f188166-b2ce-4945-beca-4670ade2aa72.png">

Reply:
looks great!